src/core/qgssnappingutils.cpp

```diff
@@ -359,7 +411,13 @@ void QgsSnappingUtils::prepareIndex( const QList<LayerAndAreaOfInterest> &layers
       QgsVectorLayer *vl = entry.first;
       QTime tt;
       tt.start();
+
+      QgsFeatureIds fids = isMatchAVisibleLayer( vl );
```

Comment:
I think here you may need to check:

   if ( mEnableSnappingForInvisibleFeature && fids.empty() )
      continue; //no visible features, skip layer

Reply:
done with a minor change
```
if ( !mEnableSnappingForInvisibleFeature && fids.empty() )
continue; //no visible features, skip layer
```